command/genesis/predeploy/params.go

```diff
@@ -0,0 +1,178 @@
+package predeploy
+
+import (
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/polygon-edge/chain"
+	"github.com/0xPolygon/polygon-edge/command"
+	"github.com/0xPolygon/polygon-edge/command/helper"
+	"github.com/0xPolygon/polygon-edge/contracts/staking"
+	"github.com/0xPolygon/polygon-edge/helper/predeployment"
+	"github.com/0xPolygon/polygon-edge/types"
+	"math/big"
+	"os"
+	"strings"
+)
+
+const (
+	chainFlag            = "chain"
+	predeployAddressFlag = "predeploy-address"
+	artifactsPathFlag    = "artifacts-path"
+	constructorArgsPath  = "constructor-args"
+)
+
+var (
+	errInvalidPredeployAddress  = errors.New("invalid predeploy address provided")
+	errAddressTaken             = errors.New("the provided predeploy address is taken")
+	errReservedPredeployAddress = errors.New("the provided predeploy address is reserved")
+	errInvalidAddress           = errors.New("the provided predeploy address must be >= 0x01100")
+)
+
+var (
+	predeployAddressMin = "0x0000000000000000000000000000000000001100"
+)
+
+var (
+	params = &predeployParams{}
+)
+
+type predeployParams struct {
+	addressRaw         string
+	constructorArgsRaw []string
+	genesisPath        string
+
+	address         types.Address
+	artifactsPath   string
+	constructorArgs []interface{}
+
+	genesisConfig *chain.Chain
+}
+
+func (p *predeployParams) getRequiredFlags() []string {
+	return []string{
+		predeployAddressFlag,
+		artifactsPathFlag,
+	}
+}
+
+func (p *predeployParams) initRawParams() error {
+	if err := p.initPredeployAddress(); err != nil {
+		return err
+	}
+
+	if err := p.verifyMinAddress(); err != nil {
+		return err
+	}
+
+	if err := p.initChain(); err != nil {
+		return err
+	}
+
+	p.convertConstructorArgs()
+
+	return nil
+}
+
+func (p *predeployParams) initPredeployAddress() error {
+	if len(p.addressRaw) < 1 {
+		return errInvalidPredeployAddress
+	}
+
+	address := types.StringToAddress(p.addressRaw)
+	if address == staking.AddrStakingContract {
+		return errReservedPredeployAddress
+	}
+
+	p.address = address
+
+	return nil
+}
+
+func (p *predeployParams) convertConstructorArgs() {
+	if len(p.constructorArgsRaw) < 1 {
```

Comment:
Why not compared with zero?

Reply:
I've dropped this method entirely as part of an earlier comment:

[b63064e](https://github.com/0xPolygon/polygon-edge/pull/536/commits/b63064e18312aa4139591c34fb8cbcf0099e8a8a)